src/components/nav/Nav.js

```diff
@@ -1,7 +1,321 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
+import styled from "styled-components";
+import { NavLink } from "react-router-dom";
+import SearchBox from "./components/searchBox/SearchBox";
+import LoginModal from "./components/loginModal/LoginModal";
+import { useHandleStatus } from "../../hooks/useHandleStatus";
 
 const Nav = () => {
-  return <div>Nav</div>;
+  const { isStatus, handleStatus } = useHandleStatus();
+  const [loginToken, setLoginToken] = useState(null);
+  const [scrollPosition, setScrollPosition] = useState(0);
+  const [hoverdMenu, setHoverdMenu] = useState(false);
+
+  const updateScroll = () => {
+    setScrollPosition(window.scrollY || document.documentElement.scrollTop);
+  };
+
+  const logout = () => {
+    localStorage.removeItem("kakao_916b3b8315bc5fed5715a2c95c878e8a");
+    window.location.reload();
+  };
+
+  useEffect(() => {
+    setLoginToken(
+      localStorage.getItem("kakao_916b3b8315bc5fed5715a2c95c878e8a")
+    );
+    window.addEventListener("scroll", updateScroll);
+  }, []);
+
+  return (
+    <>
+      <NavBar primary={scrollPosition}>
+        <NavWrapper>
+          <Logo>
+            <LogoLink primary={scrollPosition} to="/">
+              나는어때
+            </LogoLink>
+          </Logo>
+          <SearchWrapper>
+            <SearchBox />
+          </SearchWrapper>
+          <NavList>
+            <ListItemMenu>
+              <ListLink primary={scrollPosition} to="/list">
+                내주변
+              </ListLink>
+            </ListItemMenu>
+            <ListItemMenu>
+              <ListLocal
+                onMouseEnter={() => setHoverdMenu(true)}
+                onMouseLeave={() => setHoverdMenu(false)}
+                primary={scrollPosition}
+                to="/detail"
+              >
+                지역별
+                <CitiesGroup hovered={hoverdMenu}>
+                  {CITY_NAME.map(citys => (
+                    <CityName to="#" key={citys.id}>
+                      {citys.city}
+                    </CityName>
+                  ))}
+                </CitiesGroup>
+              </ListLocal>
+            </ListItemMenu>
+            <ListItemMenu>
+              <ListLink primary={scrollPosition} to="/more">
+                더보기
+              </ListLink>
+            </ListItemMenu>
+            <ListItemMenu>
+              {loginToken ? (
+                <ItemMenuButton primary={scrollPosition} onClick={logout}>
+                  로그아웃
+                </ItemMenuButton>
+              ) : (
+                <ItemMenuButton primary={scrollPosition} onClick={handleStatus}>
+                  로그인
+                </ItemMenuButton>
+              )}
```

Comment:
조건에 따라 달리지는 부분에만 삼항연산자 적용할 수 있겠네요!
```suggestion
              <ItemMenuButton primary={scrollPosition} onClick={loginToken ? logout : handleStatus}>
                {loginToken ? '로그아웃' : '로그인'}
              </ItemMenuButton>
              
```

Reply:
삼항연산자 반영했습니다 감사합니다.